app/scripts/lib/mailcheck.js

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// A ux utility to suggest correct spelling of email domains
+'use strict';
+
+define([
+  'views/tooltip',
+  'lib/url',
+  'mailcheck'
+],
+function (Tooltip, Url) {
+
+  var DOMAINS = [ // domains that get suggested, i.e gnail.com => gmail.com
+    'gmail.com', 'qq.com', 'yandex.ru', 'o2.pl', 'rambler.ru', 'googlemail.com', 't-online.de', 'mail.ru', 'web.de',
```

Comment:
I'm late to the party! It would be easier to find domains if they were on alphabetical order.